03-transactions.md

```diff
@@ -698,73 +698,408 @@ HTLC-success) gives a weight of:
+INTERNAL: local_per_commitment_point: 025f7117a78150fe2ef97db7cfc83bd57b2e2c0d0dd25eaf467a4a1c2a45ce1486
+INTERNAL: remote_per_commitment_point: 022c76692fd70814a8d1ed9dedc833318afaaed8188db4d14727e2e99bc619d325
+INTERNAL: remote_secretkey: 839ad0480cde69fc721fb8e919dcf20bc4f2b3374c7b27ff37f200ddfa7b0edb01
+# From local_delayed_payment_basepoint_secret, local_per_commitment_point and local_delayed_payment_basepoint
+INTERNAL: local_delayed_secretkey: adf3464ce9c2f230fd2582fda4c6965e4993ca5524e8c9580e3df0cf226981ad01
+-->
+
+Here are the points used to derive the obscuring factor for the commitment number:
+
+    local_payment_basepoint: 034f355bdcb7cc0af728ef3cceb9615d90684bb5b2ca5f859ab0f0b704075871aa
+    remote_payment_basepoint: 032c0b7cf95324a07d05398b240174dc0c2be444d96b159aa6c7f7b1e668680991
+	# obscured commitment transaction number = 0x2bb038521914 ^ 42
+
+And here are the keys needed to create the transactions:
+
+    local_funding_privkey: 30ff4956bbdd3222d44cc5e8a1261dab1e07957bdac5ae88fe3261ef321f37490101
+    local_funding_pubkey: 023da092f6980e58d2c037173180e9a465476026ee50f96695963e8efe436f54eb
+    remote_funding_pubkey: 030e9f7b623d2ccc7c9bd44d66d5ce21ce504c0acf6385a132cec6d3c39fa711c1
+    local_secretkey: bb13b121cdc357cd2e608b0aea294afca36e2b34cf958e2e6451a2f27469449101
+    localkey: 030d417a46946384f88d5f3337267c5e579765875dc4daca813e21734b140639e7
+    remotekey: 039390232673a9de88820d44ea910f364a332dc815cb0122bf5088d581dcbac878
+    local_delayedkey: 03fd5960528dc152014952efdb702a88f71e3c1653b2314431701ec77e57fde83c
+    local_revocation_key: 0212a140cd0c6539d07cd08dfe09984dec3251ea808b892efeac3ede9402bf2b19
+    # funding wscript = 5221023da092f6980e58d2c037173180e9a465476026ee50f96695963e8efe436f54eb21030e9f7b623d2ccc7c9bd44d66d5ce21ce504c0acf6385a132cec6d3c39fa711c152ae
+
+And here are the test vectors themselves:
 
     name: simple tx with two outputs
     to_local_msat: 7000000000
     to_remote_msat: 3000000000
-    output commit_tx: 020000000142a26bb3a430a536cf9e3a8ce2cbcb0427c29ec6c7d647175cfc78328b57fba701000000005347388002c0c62d00000000001600141844e52616af46f531635b5b770737ec5695a08b53a66a00000000002200201c1a9b14ca64510fa53ec4a910dfbf023983489f82f3f51c5884941ef0d1944234ee4e00
-
-    name: two outputs with fundee below dust limit
-    to_local_msat: 9999000000
-    to_remote_msat: 1000000
-    output commit_tx: 020000000142a26bb3a430a536cf9e3a8ce2cbcb0427c29ec6c7d647175cfc78328b57fba701000000005347388002e8030000000000001600141844e52616af46f531635b5b770737ec5695a08b2b699800000000002200201c1a9b14ca64510fa53ec4a910dfbf023983489f82f3f51c5884941ef0d1944234ee4e00
-
-    name: with htlcs, all above dust limit
-    to_local_msat: 6970000000
+    feerate_per_kw: 15000
+    # base commitment transaction fee = 10860
+    # to-local amount 6989140 wscript 63210212a140cd0c6539d07cd08dfe09984dec3251ea808b892efeac3ede9402bf2b19670190b2752103fd5960528dc152014952efdb702a88f71e3c1653b2314431701ec77e57fde83c68ac
```

Comment:
Ah, that fell out through my generic number encoding logic.  Oops, bitcoind says 0x0190 is -16, which seems valid.  Will fix my number logic.